### DISCUSSION TITLE
How to deploy the project locally?

### DISCUSSION THREAD

**Discussion by Kyle-Law** [sarahsau/indoynab#2] [Q&A]
How to deploy the project locally?

Steps I've done:
1) Cloned the repo
2) `brew install shared-mime-info` (Dependencies of mimemagic gem, referred by https://github.com/mimemagicrb/mimemagic#dependencies)
3) `bundle` to install gems
4) `rails db:create`, `rails db:migrate` to create the database (Do I need this step? Also there are no migration files btw)

Then I was stuck with this
![image](https://user-images.githubusercontent.com/55923773/113655501-855ea800-96cc-11eb-9664-070d9f515d45.png)

**Comment by Kyle-Law** [comment]
How I deploy locally (9 Apr 2021)
1.  Run `rails g model converter bank statement:binary` and type 'n' for not overwriting existing converter files
2. Run `brew install shared-mime-info` (Dependencies of mimemagic gem, referred by https://github.com/mimemagicrb/mimemagic#dependencies
3. Run `bundle install` to Install gems
4. Run `yarn install` to install javascript libraries
5. Run `rails s` and visit `localhost:3000`
6. Run ` bundle exec rails webpacker:install` if facing `Webpacker::Manifest::MissingEntryError`

**Reply by sarahsau** [reply]
Hi Kyle, yes, turns out I can replicate the Webpacker issue on the cloned repo, however the original repo that I push the code from can run normally. Investigating it now. 

For the 'table missing error', couldn't replicate the issue... only `rails db:create` is needed since there is no migration, as you mentioned.

**Reply by sarahsau** [reply]
Turns out the `public/packs` folder was included in the `.gitignore` automatically, thus not included in the remote repo. Fixed the issue by commenting out the line and committing the said folder to GitHub. Let me know if this fixes the issue at your end!
